pdc/apps/compose/views.py

```diff
@@ -1399,3 +1402,142 @@ def list(self, request):
+
+
+class ComposeTreeViewSet(ChangeSetModelMixin,
+                         StrictQueryParamMixin,
+                         MultiLookupFieldMixin,
+                         viewsets.GenericViewSet):
+    """
+    API endpoint that allows querying compose-variant-arch relevant to location.
+
+    ##Test tools##
+
+    You can use ``curl`` in terminal, with -X _method_ (GET|POST|PUT|DELETE),
+    -d _data_ (a json string). or GUI plugins for
+    browsers, such as ``RESTClient``, ``RESTConsole``.
+    """
+    queryset = ComposeTree.objects.all()
+    serializer_class = ComposeTreeSerializer
+    filter_class = ComposeTreeFilter
+    lookup_fields = (
+        ('compose__compose_id', r'[^/]+'),
+        ('variant__variant_uid', r'[^/]+'),
+        ('arch', r'[^/]+'),
+        ('location__short', r'[^/]+'),
+    )
+
+    def list(self, *args, **kwargs):
+        """
+        __Method__: GET
+
+        __URL__: $LINK:composetreelocations-list$
+
+        __Query params__:
+
+        %(FILTERS)s
+
+        __Response__: a paged list of following objects
+
+        %(SERIALIZER)s
+        """
+        return super(ComposeTreeViewSet, self).list(*args, **kwargs)
+
+    def create(self, request, *args, **kwargs):
+        """
+        __Method__: POST
+
+        __URL__: $LINK:composetreelocations-list$
+
+        __Data__:
+
+        %(WRITABLE_SERIALIZER)s
+
+         * *synced_content*: $LINK:contentdeliverycontentcategory-list$
+
+        __Response__: Same as input data.
+
+        __NOTE__:
+        If synced_content is omitted, all content types are filled in.
+        """
+        if not request.data.get("synced_content"):
+            request.data["synced_content"] = ['binary', 'debug', 'source']
+        return super(ComposeTreeViewSet, self).create(request, *args, **kwargs)
+
+    def retrieve(self, *args, **kwargs):
+        """
+        __Method__: GET
+
+        __URL__: $LINK:composetreelocations-detail:compose_id}/{variant_uid}/{arch}/{location$
+
+        __Response__:
+
+        %(SERIALIZER)s
+        """
+        return super(ComposeTreeViewSet, self).retrieve(*args, **kwargs)
+
+    def update(self, request, *args, **kwargs):
+        # This method is used by bulk update and partial update, but should not
+        # be called directly.
+        if not kwargs.get('partial', False):
+            return self.http_method_not_allowed(request, *args, **kwargs)
+
+        if not request.data:
+            return NoEmptyPatchMixin.make_response()
+
+        updatable_keys = set(['scheme', 'url', 'synced_content'])
+        if set(request.data.keys()) - updatable_keys:
+            return Response(status=status.HTTP_400_BAD_REQUEST,
+                            data={'detail': 'Only these properties can be updated: %s'
+                                  % ', '.join(updatable_keys)})
+
+        return super(ComposeTreeViewSet, self).update(request, *args, **kwargs)
+
+    def bulk_update(self, *args, **kwargs):
```

Comment:
Is it necessary to implement it? I think with BulkRouter, the default implementation is like this.